src/WinRT.Runtime/Projections/ICustomPropertyProvider.net5.cs

```diff
@@ -289,25 +385,34 @@ static unsafe ManagedCustomPropertyProviderVftbl()
 
 
         [UnmanagedCallersOnly]
-
         private static unsafe int Do_Abi_GetCustomProperty_0(IntPtr thisPtr, IntPtr name, IntPtr* result)
         {
             global::Microsoft.UI.Xaml.Data.ICustomProperty __result = default;
             try
             {
                 string _name = MarshalString.FromAbi(name);
-                object target = global::WinRT.ComWrappersSupport.FindObject<object>(thisPtr);
-                PropertyInfo propertyInfo = target.GetType().GetProperty(
-                     _name,
-                     BindingFlags.Instance | BindingFlags.Static | BindingFlags.Public);
-
-                if (propertyInfo is object)
-                {
-                    __result = new ManagedCustomProperty(propertyInfo);
+                object target = global::WinRT.ComWrappersSupport.FindObject<object>(thisPtr);
+
+                if (target is global::Microsoft.UI.Xaml.Data.IBindableCustomPropertyImplementation bindableCustomPropertyImplementation)
+                {
+                    __result = bindableCustomPropertyImplementation.GetProperty(_name);
+                    *result = MarshalInterface<global::Microsoft.UI.Xaml.Data.ICustomProperty>.FromManaged(__result);
+                    return 0;
+                }
+
+                if (RuntimeFeature.IsDynamicCodeCompiled)
+                {
+                    PropertyInfo propertyInfo = target.GetType().GetProperty(

```

Comment:
We should move this into a helper method and add the suppressions on it, right? Otherwise this will still warn on publish.

Reply:
I thought that was going to happen, but not seeing the warning strangely.  I even just fixed WarningsNotAsErrors to not include it.

Reply:
I do see is sizoscope, none of this code seems to be included.

Reply:
> "but not seeing the warning strangely"

Mmh that is weird, doesn't seem right. Have you tried with .NET 9? There's a bunch of missing warnings on .NET 8.

Reply:
I do see the warning and error on .NET 8 as well:

![image](https://github.com/user-attachments/assets/7a0b05c9-5e40-4c89-b820-57eed9bd0e91)

I really think we should do the same pattern we used elsewhere: move to a helper with the suppression, always throw on AOT, otherwise call the helper for backwards-compat.

Reply:
Strange, I don't see those.  But addressed.